### PR TITLE
Tpetra MultiVector: Remove unnecessary check from isSameSize

### PR DESCRIPTION
**Pull request by cgcgcg**
@trilinos/tpetra 

## Motivation
Remove unnecessary check from `MV::isSameSize`. This avoids a potential memcopy to host.

### CODE REVIEW

**Review by kddevin — APPROVED** [review]
Looks good to me.  The compiler should prevent multivectors with different scalar types from being compared in this function.  
I'll add the UVM Removal tag, as this issue was more costly with UVM=OFF.
Thanks @cgcgcg

File: packages/tpetra/core/src/Tpetra_MultiVector_def.hpp
Comment (kyungjoo-kim) [inline packages/tpetra/core/src/Tpetra_MultiVector_def.hpp:4695]: @kddevin @cgcgcg This PR is already merged but I have one curiosity. The type check is not really meaningful if the value is not dynamically sized. Does this condition require for Sacado DFAD, where its packing size is determined only in runtime ?

Comment (rppawlo) [reply packages/tpetra/core/src/Tpetra_MultiVector_def.hpp:4695]: @etphipp

Comment (kyungjoo-kim) [reply packages/tpetra/core/src/Tpetra_MultiVector_def.hpp:4695]: @etphipp Do you happen to know that Mark put this condition for SACADO ?

Comment (kddevin) [reply packages/tpetra/core/src/Tpetra_MultiVector_def.hpp:4695]: @kyungjoo-kim  good question.  I was expecting the compiler to enforce that v1(0,0) and v2(0,0) would have the same size, since this and vec are MultiVector<Scalar>.
@etphipp ?

Comment (etphipp) [reply packages/tpetra/core/src/Tpetra_MultiVector_def.hpp:4695]: My guess is it was for stokhos, actually which does have a dynamically sized type (we currently don't run Sacado through Tpetra, but if we did, this would also be relevant to DFad).

Comment (etphipp) [reply packages/tpetra/core/src/Tpetra_MultiVector_def.hpp:4695]: I'm not aware of any use-case that would call this function with differently sized types, however.

Comment (etphipp) [reply packages/tpetra/core/src/Tpetra_MultiVector_def.hpp:4695]: If everything passes with it removed, I don't see why you shouldn't merge it.

Comment (kyungjoo-kim) [reply packages/tpetra/core/src/Tpetra_MultiVector_def.hpp:4695]: This is already merged. ; ).

Comment (kyungjoo-kim) [reply packages/tpetra/core/src/Tpetra_MultiVector_def.hpp:4695]: Maybe it might be better if we put the check with debug ifdef guard just in case.

Comment (cgcgcg) [reply packages/tpetra/core/src/Tpetra_MultiVector_def.hpp:4695]: I'll let you guys decide what's best here. I just didn't like the fact that a host copy was triggered for a type check.

**Pull request merged**
